### PR TITLE
Dynamically calculate shrink size in Windows integration tests

### PR DESCRIPTION
**Pull request by ay901246**
### What is this change about?

On Windows stemcell `2019.103+`, disk footprint and unmovable NTFS cluster offsets on the default 30 GB root disk increased. This narrowed the gap between the partition size and `(Get-PartitionSupportedSize -DriveLetter C).SizeMin` to less than 200 MB.

In `integration/windows/environment_test.go`, `ShrinkRootPartition()` hardcoded shrinking C: by 200 MB (`const shrinkBy = 200 * MB`). When `(CurrentSize - SizeMin) < 200 MB`, `Resize-Partition` fails with:
```
Resize-Partition : Size Not Supported
Extended information:
The specified shrink size is too big and will cause the volume to be smaller than the minimum volume size.
```

This caused consistent failures in the `test-integration-windows` Concourse CI job starting in build [1065](https://ci.bosh-ecosystem.cf-app.com/teams/main/pipelines/bosh-agent-main/jobs/test-integration-windows/builds/1065). Additionally, when `ShrinkRootPartition()` failed, `AfterEach` called `EnsureAgentServiceStopped() -> CheckAgentRunning()`, which failed with exit code 1 because `Get-Service -Name bosh-agent` was invoked before the agent service was ever installed.

This PR makes two improvements in `integration/windows/environment_test.go`:
1. **Dynamic Partition Shrink:** Instead of hardcoding 200 MB, `ShrinkRootPartition()` queries `(Get-PartitionSupportedSize -DriveLetter C).SizeMin` and calculates `gap = CurrentSize - SizeMin`. It shrinks by `min(200MB, floor(gap / 2))` (requiring at least 10 MB). This guarantees the target size is strictly above `SizeMin` and frees ample space for the agent's 1 MB ephemeral partition threshold.
2. **Robust Teardown:** In `CheckAgentRunning()`, suppresses error via `-ErrorAction SilentlyContinue` and uses `RunPowershellCommandWithOffsetAndResponses` so that checking service status does not fail the test when the service is absent. In `EnsureAgentServiceStopped()`, guards `c:\bosh\service_wrapper.exe uninstall` under the running check and guards logfile deletion with `If (Test-Path ...)`.

### Please provide contextual information.

- Concourse CI pipeline: [bosh-agent-main / test-integration-windows / build 1065+](https://ci.bosh-ecosystem.cf-app.com/teams/main/pipelines/bosh-agent-main/jobs/test-integration-windows/builds/1065)
- Stemcell version: Windows `2019.103`+

### What tests have you run against this PR?

- Verified compilation of Windows integration tests: `go test -c ./integration/windows`
- Executed unit tests for Windows platform packages: `go test ./platform/windows/...`
- Validated PowerShell 5.1 syntax and edge cases for `Resize-Partition` calculation

### How should this change be described in bosh-agent release notes?

None (test suite improvement).

### Does this PR introduce a breaking change?

No.

### Tag your pair, your PM, and/or team!

@cloudfoundry/bosh-agent-reviewers

### AI Review Feedback

_All AI review comments (CodeRabbit, and Copilot when assigned) must be resolved before human reviewers will look at the PR. For each comment, either:_
- _Make the suggested change, or_
- _Reply to the comment explaining why it does not apply, then resolve the thread._

Made with [Cursor](https://cursor.com)

### CODE REVIEW

File: integration/windows/environment_test.go [hunk 1]
```diff
@@ -195,7 +199,9 @@ func (e *WindowsEnvironment) StartAgent() {
 }
 
 func (e *WindowsEnvironment) CheckAgentRunning(offset int) bool {
-	stdout := e.RunPowershellCommandWithOffset(offset+1, "Get-Service -Name bosh-agent | Format-List -Property Status")
+	stdout, _, _, _ := e.RunPowershellCommandWithOffsetAndResponses(
```
[the remark this answers is not in the document]
Comment (ay901246) [reply integration/windows/environment_test.go]: ### **_GEMINI AI_**:

Updated in commit `ee1492c45`. We now capture both `err` and `exitCode` returned by `RunPowershellCommandWithOffsetAndResponses` and assert `Expect(err).NotTo(HaveOccurred())` and `Expect(exitCode).To(BeZero())` before evaluating `stdout`. This satisfies the `errcheck` linter rule and prevents transport/command errors from being silently treated as an absent service.

File: integration/windows/environment_test.go [hunk 2]
```diff
@@ -204,9 +210,9 @@ func (e *WindowsEnvironment) CheckAgentRunning(offset int) bool {
 func (e *WindowsEnvironment) EnsureAgentServiceStopped() {
 	if e.CheckAgentRunning(1) {
 		e.RunPowershellCommandWithOffset(1, `c:\bosh\service_wrapper.exe stop`)
+		e.RunPowershellCommandWithOffset(1, `c:\bosh\service_wrapper.exe uninstall`)
```
[the remark this answers is not in the document]
Comment (ay901246) [reply integration/windows/environment_test.go:228]: ### **_GEMINI AI_**:

Updated in commit `ee1492c45`. We introduced `CheckAgentServiceExists(offset int) bool` to query service presence independently of execution state. In `EnsureAgentServiceStopped()`, the service is stopped if running, and uninstalled whenever it exists in SCM (even if stopped or crashed).

**Review by aramprice — DISMISSED** [review]

File: integration/windows/environment_test.go [hunk 3]
```diff
@@ -28,26 +28,30 @@ type WindowsEnvironment struct {
...
+		`$current = $part.Size; ` +
+		`$min = $supported.SizeMin; ` +
+		`$gap = $current - $min; ` +
+		`if ($gap -lt 10MB) { ` +
+		`    throw "Cannot shrink partition C: current size is $current, SizeMin is $min (gap is only $gap bytes, need at least 10MB)"; ` +
+		`}; ` +
+		`$shrinkBy = [int64]($gap / 2); ` +
+		`if ($shrinkBy -gt 200MB) { $shrinkBy = 200MB }; ` +
+		`$target = $current - $shrinkBy; ` +
+		`$part | Resize-Partition -Size $target`
```
Comment (aramprice) [inline integration/windows/environment_test.go]: Might be nicer to have a full text block here instead of a bunch of string + string

**Review by mkocher — DISMISSED** [review]

**Review by aramprice — APPROVED** [review]
